examples/ota-provider-app/ota-provider-common/UserConsentDelegate.h

```diff
@@ -0,0 +1,81 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/NodeId.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace ota {
+
+enum UserConsentState
+{
+    // User consent is granted.
+    kGranted,
+    // Obtaining user consent is in progress, async implementations should return this state.
+    kObtaining,
+    // User consent is denied.
+    kDenied,
+    kUnknown,
+};
+
+struct UserConsentSubject
```

Comment:
Please document what this is for and what the fields mean (which goes hand-in-hand with my comment about naming the fields to be clearer about whose vendorId it is, etc.

Reply:
Renamed the fields as per suggestion and added the description for the struct and its fields.